pkg/receive/handler.go

```diff
@@ -1025,7 +1048,7 @@ func (es errorSet) Error() string {
 
 // Add adds an error to the errorSet.
 func (es *errorSet) Add(err error) {
-	if err == nil {
+	if err == nil || isConflict(err) {
```

Comment:
this line doesn't add isconlict error to the error set at all

Reply:
is this the first stop the error got returned from grpc? I am worried the error was surfaced in other places too.

Reply:
you should make err = nil right after grpc call here:
```
			// Create a span to track the request made to another receive node.
			tracing.DoInSpan(fctx, "receive_forward", func(ctx context.Context) {
				// Actually make the request against the endpoint we determined should handle these time series.
				_, err = cl.RemoteWrite(ctx, &storepb.WriteRequest{
					Timeseries: wreqs[writeTarget].timeSeries,
					Tenant:     tenant,
					// Increment replica since on-the-wire format is 1-indexed and 0 indicates un-replicated.
					Replica: int64(writeTarget.replica + 1),
				})
			})
```